links/src/main/java/juja/microservices/links/repository/impl/LinksRepositoryImpl.java

```diff
@@ -51,6 +51,10 @@ public Link getLinkByURL(String url) throws NotFoundException {
 
     @Override
     public List<Link> getAllLinks() {
-        return mongoTemplate.findAll(Link.class, mongoCollectionName);
+        Query query = new Query();
+        query.addCriteria(Criteria.where("hidden").is(false));
+        List<Link> result = mongoTemplate.find(query, Link.class, mongoCollectionName);
+        log.info("Found {} active links in database", result);
+        return result;
```

Comment:
i doubt that this is a good idea to apply filters to getAllLinks() method at repository level. 
what about a new one like getAllNotHiddenLinks or anything similar?

Reply:
Exactly. Thanks